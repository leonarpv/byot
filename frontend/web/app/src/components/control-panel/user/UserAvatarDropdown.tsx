import * as React from 'react';
import {
  Avatar,
  Box,
  ButtonBase,
  Divider,
  makeStyles,
  Menu,
  MenuItem,
  Theme,
  ListItemIcon,
} from '@material-ui/core';
import {KeyboardArrowDown, ExitToApp, VerifiedUser} from '@material-ui/icons';
import {WithStyles} from '@byot-frontend/web-common/src/types/WithStyles';
import {Link} from 'react-router-dom';
import {useTranslation} from 'react-i18next';

interface Props extends WithStyles<typeof styles> {
  onLogout?: () => void;
}

const styles = (theme: Theme) => ({
  button: {
    borderRadius: theme.shape.borderRadius,
  },
});
const useStyles = makeStyles(styles);

export const UserAvatarDropdown: React.FC<Props> = (props: Props) => {
  const styles = useStyles(props);
  const {t} = useTranslation();
  const [anchorElement, setAnchorElement] = React.useState<Element | null>(null);
  const setMenuOpen = (e: React.MouseEvent<HTMLButtonElement>) => {
    setAnchorElement(e.currentTarget);
  };
  const setMenuClose = () => {
    setAnchorElement(null);
  };
  return (
    <>
      <Box px={2}>
        <ButtonBase
          data-testid="control-panel-user-avatar-dropdown-trigger"
          classes={{root: styles.button}}
          onClick={setMenuOpen}>
          <Box pr={1} display="flex" flexDirection="row" alignItems="center">
            <Avatar />
            <KeyboardArrowDown />
          </Box>
        </ButtonBase>
      </Box>
      <Menu
        open={!!anchorElement}
        data-testid="control-panel-user-avatar-dropdown-menu"
        onClose={setMenuClose}
        anchorEl={anchorElement}
        getContentAnchorEl={null}
        anchorOrigin={{vertical: 'bottom', horizontal: 'right'}}
        transformOrigin={{vertical: 'top', horizontal: 'right'}}>
        <MenuItem component={Link} to="/">
          <ListItemIcon>
            <VerifiedUser fontSize="small" />
          </ListItemIcon>
          {t('Profile')}
        </MenuItem>
        <Divider />
        <MenuItem onClick={props.onLogout}>
          <ListItemIcon>
            <ExitToApp fontSize="small" />
          </ListItemIcon>
          {t('Log out')}
        </MenuItem>
      </Menu>
    </>
  );
};

/*
export const UserAvatarDropdownRedux: React.FC<Partial<Props>> = (props: Partial<Props>) => {
    const dispatch = useDispatch();
    const onLogout = () => dispatch(ProcessActionExtractor.dispatch(Logout, {}))
    return <UserAvatarDropdown {...props as Required<Props>} onLogout={onLogout}/>;
}
*/
