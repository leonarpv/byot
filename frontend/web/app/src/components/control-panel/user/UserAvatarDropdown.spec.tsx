import * as React from 'react';
import {render, fireEvent} from '@testing-library/react';
import {UserAvatarDropdown} from './UserAvatarDropdown';

describe('<UserAvatarDropdown/>', () => {
  it('should render', () => {
    const {container, queryByTestId} = render(<UserAvatarDropdown />);
    expect(queryByTestId('control-panel-user-avatar-dropdown-menu')).toBeNull();
    expect(container).toMatchSnapshot();
  });
  it('should render and open modal', () => {
    const {baseElement, queryByTestId} = render(<UserAvatarDropdown />);
    const trigger = queryByTestId('control-panel-user-avatar-dropdown-trigger');
    expect(trigger).not.toBeNull();
    fireEvent.click(trigger!);
    expect(queryByTestId('control-panel-user-avatar-dropdown-menu')).not.toBeNull();
    expect(baseElement).toMatchSnapshot();
  });
});
