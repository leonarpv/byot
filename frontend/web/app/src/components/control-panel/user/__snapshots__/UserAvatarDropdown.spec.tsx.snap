// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserAvatarDropdown/> should render 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-2"
  >
    <button
      class="MuiButtonBase-root makeStyles-button-1"
      data-testid="control-panel-user-avatar-dropdown-trigger"
      tabindex="0"
      type="button"
    >
      <div
        class="MuiBox-root MuiBox-root-3"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiAvatar-fallback"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
            />
          </svg>
        </div>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
          />
        </svg>
      </div>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;

exports[`<UserAvatarDropdown/> should render and open modal 1`] = `
<body
  style="padding-right: 0px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  >
    <div
      class="MuiBox-root MuiBox-root-5"
    >
      <button
        class="MuiButtonBase-root makeStyles-button-4"
        data-testid="control-panel-user-avatar-dropdown-trigger"
        tabindex="0"
        type="button"
      >
        <div
          class="MuiBox-root MuiBox-root-6"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiAvatar-fallback"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
              />
            </svg>
          </div>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
            />
          </svg>
        </div>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
  <div
    class="MuiPopover-root"
    data-testid="control-panel-user-avatar-dropdown-menu"
    role="presentation"
    style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
  >
    <div
      aria-hidden="true"
      style="z-index: -1; position: fixed; right: 0px; bottom: 0px; top: 0px; left: 0px; background-color: transparent;"
    />
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiPaper-root MuiMenu-paper MuiPopover-paper MuiPaper-elevation8 MuiPaper-rounded"
      style="opacity: 1; transform: scale(1, 1); transition: opacity 0ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 0ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; top: 16px; left: 16px; transform-origin: -16px -16px;"
      tabindex="-1"
    >
      <ul
        class="MuiList-root MuiMenu-list MuiList-padding"
        role="menu"
        tabindex="-1"
      >
        <li
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button Mui-focusVisible Mui-focusVisible"
          role="menuitem"
          tabindex="0"
          to="/"
        >
          <div
            class="MuiListItemIcon-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 1L3 5v6c0 5.55 3.84 10.74 9 12 5.16-1.26 9-6.45 9-12V5l-9-4zm-2 16l-4-4 1.41-1.41L10 14.17l6.59-6.59L18 9l-8 8z"
              />
            </svg>
          </div>
          Profile
          <span
            class="MuiTouchRipple-root"
          />
        </li>
        <hr
          class="MuiDivider-root"
        />
        <li
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
          role="menuitem"
          tabindex="-1"
        >
          <div
            class="MuiListItemIcon-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10.09 15.59L11.5 17l5-5-5-5-1.41 1.41L12.67 11H3v2h9.67l-2.58 2.59zM19 3H5c-1.11 0-2 .9-2 2v4h2V5h14v14H5v-4H3v4c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
          </div>
          Log out
          <span
            class="MuiTouchRipple-root"
          />
        </li>
      </ul>
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;
